test/Object/rlexe.xml

```diff
@@ -450,4 +450,24 @@
       <baseline>assign.baseline</baseline>
     </default>
   </test>
+  <test>
+    <default>
+      <files>ObjectSpread_Simple.js</files>
+      <compile-flags>-args summary -endargs -NoNative -ES2018ObjectSpread</compile-flags>
+    </default>
+  </test>
+  <test>
+    <default>
+      <files>ObjectSpread_JIT.js</files>
+      <compile-flags>-args summary -endargs -ES2018ObjectSpread -bgjit -maxinterpretcount:1 -off:simplejit</compile-flags>
+    </default>
+  </test>
+  <test>
+    <default>
+      <files>ObjectSpread_Limits.js</files>
+      <compile-flags>-args summary -endargs -ES2018ObjectSpread</compile-flags>
+      <tags>Slow</tags>
+      <timeout>3600</timeout>
```

Comment:
this varies a lot on the nightly machine. Our nightly machines are way slower so this may timeout regularly.

Reply:
Perhaps the test could stay but as exclude_nightly? Both Jenkins and ClassLab run slow tests, theoretically with less variance?

Reply:
We don't want this to run on ClassLab - this will slowdown the classlab significantly.

---
In reply to: [198910567](https://github.com/Microsoft/ChakraCore/pull/5268#discussion_r198910567) [](ancestors = 198910567)

Reply:
So would it be best to remove it or lengthen the timeout?

---
In reply to: [198914573](https://github.com/Microsoft/ChakraCore/pull/5268#discussion_r198914573) [](ancestors = 198914573,198910567)

Reply:
I don't see a point of this test - the max number is not near the array limit (2^31). I suggest to remove this.
or you can change the test something like
```
function foo(start, end) {
let arr = [] ;
for (i = start; i < end; i++) {
    arr[i] = i;
}
let obj = {...arr};
//assert.areEqual(maxSize, Object.keys(obj).length);
for(var propName in obj) {
  print(propName);
    propValue = obj[propName];
    //assert.areEqual(propName, propValue.toString());
}
};
foo(2**31-100, 2**31+100);
foo(2**32-100, 2**32+100);
```

---
In reply to: [198918459](https://github.com/Microsoft/ChakraCore/pull/5268#discussion_r198918459) [](ancestors = 198918459,198914573,198910567)

Reply:
I think the original point was to test spread using the largest possible Object allocation (which is only about 2GB). @digitalinfinity is this test still worth keeping?

I'll also go ahead and add this new test just for good measure. Thanks!

---
In reply to: [198947945](https://github.com/Microsoft/ChakraCore/pull/5268#discussion_r198947945) [](ancestors = 198947945,198918459,198914573,198910567)